src/test/test-execute.c

```diff
@@ -209,7 +258,7 @@ int main(int argc, char *argv[]) {
         }
 
         assert_se(setenv("XDG_RUNTIME_DIR", "/tmp/", 1) == 0);
-        assert_se(set_unit_path(TEST_DIR) >= 0);
+        assert_se(set_unit_path(TEST_DIR "/test-execute/") >= 0);
```

Comment:
I never set SYSTEMD_UNIT_PATH so I never got this problem, but I reproduced it just now. Thanks